sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -113,6 +113,8 @@ class SparkSession private(
 
   private[sql] val sessionUUID: String = UUID.randomUUID.toString
 
+  private[sql] val jobArtifactState: JobArtifactState = JobArtifactState(sessionUUID, None)
```

Comment:
Just curious. `JobArtifactState` is currently only used in Spark Connect. How do you use this with Kyuubi?

Reply:
The change of this pr is try to reuse JobArtifactState in normal Spark sql query.
set active jobArtifactState in each spark session make sure get right session uuid in method JobArtifactSet.getActiveOrDefault, then get isolated session aritifact set of each session.

In kyuubi group share level, multi user may load their own udf jar, and may exist same udf class name in different jar, lead the first load class take effect always

Reply:
"In kyuubi group share level, multi user may load their own udf jar, and may exist same udf class name in different jar, lead the first load class take effect always"
each user create their own spark session in same spark engine(spark cluster), each spark session use isolated aritifact set